examples/autotls/main.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/caddyserver/certmagic"
+	"github.com/ipfs/go-log/v2"
+
+	p2pforge "github.com/ipshipyard/p2p-forge/client"
+	"github.com/libp2p/go-libp2p"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	quic "github.com/libp2p/go-libp2p/p2p/transport/quic"
+	"github.com/libp2p/go-libp2p/p2p/transport/tcp"
+	ws "github.com/libp2p/go-libp2p/p2p/transport/websocket"
+)
+
+var logger = log.Logger("example")
+
+const userAgent = "go-libp2p/example/autotls"
+
+func main() {
+	// Create a background context
+	ctx := context.Background()
+
+	log.SetLogLevel("*", "error")
+	log.SetLogLevel("example", "debug")   // Set the log level for the example to debug
+	log.SetLogLevel("basichost", "info")  // Set the log level for the basichost package to info
+	log.SetLogLevel("autotls", "debug")   // Set the log level for the autotls-example package to debug
+	log.SetLogLevel("p2p-forge", "debug") // Set the log level for the p2pforge package to debug
+	log.SetLogLevel("nat", "debug")       // Set the log level for the libp2p nat package to debug
+
+	certLoaded := make(chan bool, 1) // Create a channel to signal when the cert is loaded
+
+	// TODO: this should not be necessary once https://github.com/ipshipyard/p2p-forge/issues/8 is resolved
+	rawLogger := logger.Desugar()
+	certmagic.Default.Logger = rawLogger.Named("default_fixme")
+	certmagic.DefaultACME.Logger = rawLogger.Named("default_acme_client_fixme")
+
+	// p2pforge is the AutoTLS client library.
+	// The cert manager handles the creation and management of certificate
+	certManager, err := p2pforge.NewP2PForgeCertMgr(
+		// Configure CA ACME endpoint
+		// NOTE:
+		// This example uses Let's Encrypt staging CA (p2pforge.DefaultCATestEndpoint)
+		// which will not work correctly in browser, but is useful for initial testing.
+		// Production should use Let's Encrypt production CA (p2pforge.DefaultCAEndpoint).
+		p2pforge.WithCAEndpoint(p2pforge.DefaultCATestEndpoint), // test CA endpoint
+		// TODO: p2pforge.WithCAEndpoint(p2pforge.DefaultCAEndpoint),  // production CA endpoint
+
+		// Configure where to store certificate
+		p2pforge.WithCertificateStorage(&certmagic.FileStorage{Path: "p2p-forge-certs"}),
```

Comment:
:information_source: added this to example, as it quite common for people to wonder  how to control this